### ISSUE TITLE
failed to notify systemd for readiness

### ISSUE THREAD

**Issue by crawford** [coreos/etcd#3613]
I see this error in my logs:

```
Sep 29 17:07:49 localhost etcd2[705]: added local member aeb6950c050a83f3 [http://10.0.0....e14
Sep 29 17:07:49 localhost etcd2[705]: failed to notify systemd for readiness
Sep 29 17:07:49 localhost etcd2[705]: aeb6950c050a83f3 is starting a new election at term 1
Sep 29 17:07:49 localhost etcd2[705]: aeb6950c050a83f3 became candidate at term 2
Sep 29 17:07:49 localhost etcd2[705]: aeb6950c050a83f3 received vote from aeb6950c050a83f...m 2
Sep 29 17:07:49 localhost etcd2[705]: aeb6950c050a83f3 became leader at term 2
Sep 29 17:07:49 localhost etcd2[705]: raft.node: aeb6950c050a83f3 elected leader aeb6950c...m 2
Sep 29 17:07:49 localhost etcd2[705]: setting up the initial cluster version to 2.2
Sep 29 17:07:50 localhost etcd2[705]: set the initial cluster version to 2.2
```

**Comment by ekozan** [comment]
hey, same error here, are you trying to set-up an single node cluster ?

**Comment by crawford** [comment]
Yes.

**Comment by ekozan** [comment]
rolf i don't understand our mistake 

```
Sep 29 18:57:45 kumsufi_test systemd[1]: Started etcd2.
Sep 29 18:57:46 kumsufi_test etcd2[11749]: recognized and used environment variable ETCD_ADVERTISE_CLIENT_URLS=http://192.168.0.27:2379
Sep 29 18:57:46 kumsufi_test etcd2[11749]: recognized and used environment variable ETCD_DATA_DIR=/var/lib/etcd2
Sep 29 18:57:46 kumsufi_test etcd2[11749]: recognized and used environment variable ETCD_INITIAL_ADVERTISE_PEER_URLS=http://192.168.0.27:2380
Sep 29 18:57:46 kumsufi_test etcd2[11749]: recognized and used environment variable ETCD_INITIAL_CLUSTER=kumsufi_test=http://192.168.0.27:2380
Sep 29 18:57:46 kumsufi_test etcd2[11749]: recognized and used environment variable ETCD_LISTEN_CLIENT_URLS=http://127.0.0.1:4001,http://127.0.0.1:2379,http://192.168.0.27:4001
Sep 29 18:57:46 kumsufi_test etcd2[11749]: recognized and used environment variable ETCD_LISTEN_PEER_URLS=http://192.168.0.27:2380,http://192.168.0.27:7001
Sep 29 18:57:46 kumsufi_test etcd2[11749]: recognized and used environment variable ETCD_NAME=kumsufi_test
Sep 29 18:57:46 kumsufi_test etcd2[11749]: etcd Version: 2.2.0
Sep 29 18:57:46 kumsufi_test etcd2[11749]: Git SHA: e4561dd
Sep 29 18:57:46 kumsufi_test etcd2[11749]: Go Version: go1.4.2
Sep 29 18:57:46 kumsufi_test etcd2[11749]: Go OS/Arch: linux/amd64
Sep 29 18:57:46 kumsufi_test etcd2[11749]: setting maximum number of CPUs to 1, total number of available CPUs is 1
Sep 29 18:57:46 kumsufi_test etcd2[11749]: the server is already initialized as member before, starting as etcd member...
Sep 29 18:57:46 kumsufi_test etcd2[11749]: listening for peers on http://192.168.0.27:2380
Sep 29 18:57:46 kumsufi_test etcd2[11749]: listening for peers on http://192.168.0.27:7001
Sep 29 18:57:46 kumsufi_test etcd2[11749]: listening for client requests on http://127.0.0.1:2379
Sep 29 18:57:46 kumsufi_test etcd2[11749]: listening for client requests on http://127.0.0.1:4001
Sep 29 18:57:46 kumsufi_test etcd2[11749]: listening for client requests on http://192.168.0.27:4001
Sep 29 18:57:46 kumsufi_test etcd2[11749]: name = kumsufi_test
Sep 29 18:57:46 kumsufi_test etcd2[11749]: data dir = /var/lib/etcd2
Sep 29 18:57:46 kumsufi_test etcd2[11749]: member dir = /var/lib/etcd2/member
Sep 29 18:57:46 kumsufi_test etcd2[11749]: heartbeat = 100ms
Sep 29 18:57:46 kumsufi_test etcd2[11749]: election = 1000ms
Sep 29 18:57:46 kumsufi_test etcd2[11749]: snapshot count = 10000
Sep 29 18:57:46 kumsufi_test etcd2[11749]: advertise client URLs = http://192.168.0.27:2379
Sep 29 18:57:47 kumsufi_test etcd2[11749]: restarting member ce2a822cea30bfca in cluster 7e27652122e8b2ae at commit index 2593
Sep 29 18:57:47 kumsufi_test etcd2[11749]: ce2a822cea30bfca became follower at term 7
Sep 29 18:57:47 kumsufi_test etcd2[11749]: newRaft ce2a822cea30bfca [peers: [], term: 7, commit: 2593, applied: 0, lastindex: 2593, lastterm: 7]
Sep 29 18:57:47 kumsufi_test etcd2[11749]: starting server... [version: 2.2.0, cluster version: to_be_decided]
Sep 29 18:57:47 kumsufi_test etcd2[11749]: added local member ce2a822cea30bfca [http://localhost:2380 http://localhost:7001] to cluster 7e27652122e8b2ae
Sep 29 18:57:47 kumsufi_test etcd2[11749]: set the initial cluster version to 2.2
Sep 29 18:57:47 kumsufi_test etcd2[11749]: failed to notify systemd for readiness
Sep 29 18:57:48 kumsufi_test etcd2[11749]: ce2a822cea30bfca is starting a new election at term 7
Sep 29 18:57:48 kumsufi_test etcd2[11749]: ce2a822cea30bfca became candidate at term 8
Sep 29 18:57:48 kumsufi_test etcd2[11749]: ce2a822cea30bfca received vote from ce2a822cea30bfca at term 8
Sep 29 18:57:48 kumsufi_test etcd2[11749]: ce2a822cea30bfca became leader at term 8
Sep 29 18:57:48 kumsufi_test etcd2[11749]: raft.node: ce2a822cea30bfca elected leader ce2a822cea30bfca at term 8
Sep 29 18:57:49 kumsufi_test etcd2[11749]: published {Name:kumsufi_test ClientURLs:[http://192.168.0.27:2379]} to cluster 7e27652122e8b2ae
```

**Comment by xiang90** [comment]
@crawford @ekozan It actually works for me. I think you need to add 
```
Type=notify
```
into your systemd service file. 
I will let etcd print out more information when it failed to send readiness to systemd too.
